### PR TITLE
use create application mutation

### LINKED ISSUE #3448: Clicking "apply to pool" creates a draft application
image

# Acceptance Criteria
- [x] Calls the create application mutation
- [x] Navigates to the the My Application page
- [x] If operation fails toast is displayed

### PR DESCRIPTION
**Pull request by vd1992**
closes #3448 
makes the application creation process work

**testing**: 
go to `/browse/pools` and click on a pool, append `/view` to the end of the URL (issue has already been created for this #3899)
if pool cannot be applied to, may need to try another pool or check its status in `/admin/pools` and publish it

click button to apply and ensure 
- success toast
- navigates to review application page
- application visible in My Applications
- Application is DRAFT

### CODE REVIEW

File: frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx [hunk 1]
```diff
@@ -126,11 +135,36 @@ const PoolAdvertisement = ({ poolAdvertisement }: PoolAdvertisementProps) => {
...
+      .then((result) => {
+        if (result.data?.createApplication) {
+          navigate(paths.reviewApplication(result.data.createApplication.id));
+          toast(
+            intl.formatMessage({
+              defaultMessage: "Application created",
+              id: "U/ji+A",
+              description: "Application created successfully",
+            }),
+          );
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx:150]: Update this to `toast.success` to match others.

```suggestion
          toast.success(
            intl.formatMessage({
              defaultMessage: "Application created",
              id: "U/ji+A",
              description: "Application created successfully",
            }),
          );
```

File: frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx [hunk 2]
```diff
@@ -126,11 +135,36 @@ const PoolAdvertisement = ({ poolAdvertisement }: PoolAdvertisementProps) => {
...
+        return result.data?.createApplication;
+      })
+      .catch((result) => {
+        toast(
+          intl.formatMessage({
+            defaultMessage: "Error application creation failed",
+            id: "tlAiJm",
+            description: "Application creation failed",
+          }),
+        );
```
Comment (esizer) [inline frontend/talentsearch/src/js/components/pool/PoolAdvertisementPage.tsx:162]: Similar to the success toast but for error styling.

```suggestion
        toast.error(
          intl.formatMessage({
            defaultMessage: "Error application creation failed",
            id: "tlAiJm",
            description: "Application creation failed",
          }),
        );
```

**Review by esizer — APPROVED** [review]
Cool! Just two suggestions for toast styling.

> Note: Accidentally hit approve instead of "Request Changes"

**Review by esizer — CHANGES_REQUESTED** [review]
Fixed review

**Comment by vd1992** [comment]
> Fixed review

and added in with https://github.com/GCTC-NTGC/gc-digital-talent/pull/3939/commits/0083467b9c395d24687ff4fc355ddc276a1e7f8c
thanks, oops I missed that styling

**Review by esizer — APPROVED** [review]
Awesome! 🧑‍🍳

**Pull request merged**